shell/common/shell.cc

```diff
@@ -283,42 +300,76 @@ std::unique_ptr<Shell> Shell::Create(
     const Shell::CreateCallback<PlatformView>& on_create_platform_view,
     const Shell::CreateCallback<Rasterizer>& on_create_rasterizer,
     DartVMRef vm) {
+  auto shell_holder_future = Shell::InitShellEnv(
+      std::move(task_runners), window_data, settings,
+      std::move(isolate_snapshot), std::move(on_create_platform_view),
+      std::move(on_create_rasterizer), std::move(vm));
+  auto shell_holder = shell_holder_future.get();
+  if (!shell_holder) {
+    return nullptr;
+  }
+
+  fml::AutoResetWaitableEvent latch;
+  std::unique_ptr<Shell> shell;
+  fml::TaskRunner::RunNowOrPostTask(
+      task_runners.GetPlatformTaskRunner(),
+      fml::MakeCopyable([&latch, &shell, &shell_holder
+
+  ]() mutable {
+        shell = Shell::MakeShellFromHolder(std::move(shell_holder));
+        latch.Signal();
+      }));
+  latch.Wait();
+
+  return shell;
+}
+
+std::future<std::unique_ptr<Shell::ShellHolder>> Shell::InitShellEnv(
```

Comment:
Nit: `InitializeShellHolder` or `CreateShellHolder` might be a better name than `InitShellEnv` as we usually don't encourage abbreviations like `Env`.